packages/Main/src/Renderer/WebGLComposer.ts

```diff
@@ -0,0 +1,181 @@
+ * @param quadCam - The camera for the quad scene.
+ */
+function drawTextureLayer(
+    renderer: THREE.WebGLRenderer,
+    renderTarget: THREE.WebGLRenderTarget,
+    dataArrayTextureToPopulate: THREE.DataArrayTexture,
+    layerIndex: number,
+    quadScene: THREE.Scene,
+    quadCam: THREE.OrthographicCamera,
+) {
+    const gl = renderer.getContext();
+
+    // 1. Set the render target.
+    const current = renderer.getRenderTarget();
+    renderer.setRenderTarget(renderTarget);
+
+    // 2. Get the raw WebGLTexture object for the DataArrayTexture.
+    const props = renderer.properties.get(dataArrayTextureToPopulate) as
+        { __webglTexture: WebGLTexture };
+    const dataArrayTextureWebGL = props.__webglTexture;
+
+    // 3. Attach the specific layer of the DataArrayTexture to the framebuffer's
+    // COLOR_ATTACHMENT0. The framebuffer is already bound by
+    // `renderer.setRenderTarget(renderTarget)`.
+    if ('framebufferTextureLayer' in gl) {
+        gl.framebufferTextureLayer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0,
+            dataArrayTextureWebGL, 0, layerIndex);
+    } else {
+        console.error('framebufferTextureLayer function not supported');
+        return;
+    }
+
+    // 4. Check framebuffer status after attaching the layer.
+    const status = gl.checkFramebufferStatus(gl.FRAMEBUFFER);
+    if (status !== gl.FRAMEBUFFER_COMPLETE) {
+        console.error('Framebuffer not complete after attaching layer:', status);
+        return;
+    }
+
+    // 5. Render the quad scene.
+    renderer.render(quadScene, quadCam);
+
+    // 6. Reset render target to null. This unbinds the custom framebuffer
+    // and restores the default framebuffer (usually the screen).
+    renderer.setRenderTarget(current);
+}
+
+/**
+ * Initializes a THREE.DataArrayTexture with immutable storage and populates
+ * its layers by rendering individual 2D textures onto them using the GPU.
+ *
+ * @param uTextures - The uniform object containing the
+ * THREE.DataArrayTexture to be initialized and populated.
+ * @param width - The width of each layer in the DataArrayTexture.
+ * @param height - The height of each layer in the DataArrayTexture.
+ * @param count - The total number of layers the DataArrayTexture should have.
+ * @param tiles - An array of RasterTile objects, each containing textures.
+ * @param max - The maximum allowed number of layers for the DataArrayTexture.
+ * @returns True if the function succeeded
+ */
+export function makeDataArrayTexture(
+    uTextures: THREE.IUniform, width: number, height: number, count: number, tiles: RasterTile[],
+    max: number,
+): boolean {
+    // If no textures found, dispose previous DataArrayTexture and reset
+    if (count === 0) {
+        if (uTextures.value instanceof THREE.DataArrayTexture) { uTextures.value.dispose(); }
+        uTextures.value = new THREE.DataArrayTexture();
+        return false;
+    }
+
+    const renderer: THREE.WebGLRenderer = view.renderer;
```

Comment:
This is a reference to a global variable. What's the best way to access the renderer without it?

Reply:
Good solution. And it does fix the minimap!